### PR TITLE
Add DatoCMS provider

### LINKED ISSUE #164: 🧩 App: CMS - DatoCMS integration

### PR DESCRIPTION
**Pull request by orzechdev**
Closes https://github.com/saleor/apps/issues/164

Completed:
- [x] Added DatoCMS provider
- [x] Updated Contribution guide
- [x] Tested manually

Example with basic configuration (with required token and item type id):

Screenshot 2023-02-18 at 18 46 30

https://user-images.githubusercontent.com/9825562/219777130-984fa7b9-ea20-404a-a0c7-7e686f438de4.mp4

### CODE REVIEW

**Review by lkostrowski — APPROVED** [review]
Code looks OK. @peelar may have more comments since he's more into that codebase

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -39,25 +43,38 @@ export const providersConfig = {
 > - generate an integration configuration view (see: `src/views/configuration.tsx`)
 > - store & fetch the tokens from the settings API (see: `src/pages/api/settings.ts`)
 
+3. Add configurartion fields to schema:
```
Comment (peelar) [inline CONTRIBUTING.md]:
```suggestion
3. Add configuration fields to schema:
```

File: src/lib/cms/types.ts [hunk 2]
```diff
@@ -16,8 +16,8 @@ export type CmsOperations = {
   getAllProducts?: () => Promise<Response>;
   getProduct?: ({ id }: { id: string }) => Promise<Response>;
   createProduct: ({ input }: { input: ProductInput }) => Promise<CreateProductResponse>;
-  updateProduct: ({ id, input }: { id: string; input: ProductInput }) => Promise<Response>;
-  deleteProduct: ({ id }: { id: string }) => Promise<Response>;
+  updateProduct: ({ id, input }: { id: string; input: ProductInput }) => Promise<Response | void>;
+  deleteProduct: ({ id }: { id: string }) => Promise<Response | void>;
```
Comment (peelar) [inline src/lib/cms/types.ts:20]: I think we can change all these (besides `createProduct`) return types to Promise<any>. That's not important for this review, we can do it later.

**Review by peelar — APPROVED** [review]
Great stuff 👏

**Pull request merged**